components/device-mgt-extensions/org.wso2.carbon.device.mgt.extensions.device.type.deployer/src/main/java/org/wso2/carbon/device/mgt/extensions/device/type/deployer/template/DeviceTypeManagerService.java

```diff
@@ -113,8 +113,12 @@ private void populatePushNotificationConfig(PushNotificationProvider pushNotific
         if (pushNotificationProvider != null) {
             if (pushNotificationProvider.isFileBasedProperties()) {
                 Map<String, String> staticProps = new HashMap<>();
-                for (Property property : pushNotificationProvider.getConfigProperties().getProperty()) {
-                    staticProps.put(property.getName(), property.getValue());
+                if (pushNotificationProvider.getConfigProperties() != null &&
+                        pushNotificationProvider.getConfigProperties().getProperty() != null &&
+                        pushNotificationProvider.getConfigProperties().getProperty().size() > 0) {
+                    for (Property property : pushNotificationProvider.getConfigProperties().getProperty()) {
```

Comment:
shall we move this pushNotificationProvider.getConfigProperties().getProperty() into a variable? We have accessed the same three times.